src/webview/webkitgtk/web_context.rs

```diff
@@ -317,11 +321,39 @@ where
           }
         }
 
+        // Set request http method
         if let Some(method) = request.http_method() {
           http_request = http_request.method(method.as_str());
         }
+
+        // Set request http body
```

Comment:
should probably be behind another feature flag `linux-body`

Reply:
I'm considering removing these flags and checking the version at runtime instead.
We encountered so many webkitgtk version difficulties. I think we should simplify the feature flag for users.
wdyt? cc @lucasfernog

Reply:
sounds good

Reply:
would be amazing

Reply:
I agree it would be super useful, as long as we can still compile the app against older webkitgtk version. I think we will need it for the IPC anyway if we really end up implementing version specific implementations like Jonas talked about.

Reply:
Right, last time we had to add feature flags for the headers because we couldn't compile against older webkit2gtk due to linking issues I think. @wusyong

Reply:
Like this is problematic https://github.com/tauri-apps/wry/actions/runs/5222821122/jobs/9428823012?pr=969#step:10:235

Reply:
damn. I'll add linux-body flag then.

Reply:
But I think we could remove linux-headers flag right? Since it's in 2.36. And we default to 2.38 now.

Reply:
In this specific case we may not need a feature flag. With the switch to webkitgtk 4.1 we raised our minium version to 2.38 basically measured by ubuntu 22.04, the first ubuntu release to include webkitgtk 4.1 - it received the update to 2.40 too so imho we _could_ require that version if it makes our lives easier.

For what it's worth, the first debian release to include 4.1 is bookworm, released 2(?) days ago which came with 2.40 out of the box.

fedora 37 also got the 2.40 update and idk what other non-rolling distros may matter.

Reply:
We can enable everything right now and if we see a problem, mark it as feature flags enabled by default that can be disabled by users. We can still use this final alpha / beta time to check it. @wusyong

Reply:
@FabianLars actually only ubuntu 23.04 has the 2.40 update it seems. https://packages.ubuntu.com/search?keywords=webkit2gtk

Reply:
I can do something crazy on the Tauri side. Run the current IPC on Linux, and if the response is a blob, send a message to JS telling "hey, use a fetch for custom protocol url `ipc://localhost/$requestId` to get the blob response". At least reading a binary file wouldn't take a minute.

Reply:
yeah, that'd be really cool! Or tbh i'd explore it for non-blobs too, maybe check if it's above 1kb or something (if not blob) because postMessage isn't exactly fast for large strings either.

Reply:
> yeah, that'd be really cool! Or tbh i'd explore it for non-blobs too, maybe check if it's above 1kb or something (if not blob) because postMessage isn't exactly fast for large strings either.

@FabianLars
https://github.com/tauri-apps/tauri/pull/7170/files#diff-cfab6e0634dd785c7949c059698552e4e8f6139ccbcdb29aeb6b4c29d57805b7R1738 
https://github.com/tauri-apps/tauri/pull/7170/files#diff-cfab6e0634dd785c7949c059698552e4e8f6139ccbcdb29aeb6b4c29d57805b7R1748
https://github.com/tauri-apps/tauri/pull/7170/files#diff-f5b23dca1c1951b8661b16b558998059ed8f2029ec6806392dcd430ddec82f02R74

for now i'm just using the protocol for any object/array/raw data. I could check the array and object length, but it's hard to determine the proper threshold.